swagger_spec_validator/util.py

```diff
@@ -48,6 +59,153 @@ def validate_spec_url(spec_url):
+            _, schema = object_mappings_resolver.resolve(json_ref['$ref'])
+            self._mappings[name] = schema
+
+    def __init_resolver(self):
+        """Init schema resolver"""
+        schema = read_file(self._schema_path)
+
+        self._resolver = RefResolver(
+            base_uri='file://{0}'.format(self._schema_path),
+            referrer=schema,
+        )
+
+    @property
+    def mappings(self):
+        if self._mappings is None:
+            self.__load_attributes()
+        return self._mappings
+
+    @property
+    def resolver(self):
+        if self._resolver is None:
+            self.__init_resolver()
+        return self._resolver
+
+    def __iter__(self):
+        return iterkeys(self.mappings)
+
+    def __contains__(self, swagger_object_type):
+        return swagger_object_type in self.mappings
+
+    def __getattr__(self, swagger_object_type):
+        if swagger_object_type in self:
+            return swagger_object_type
+
+        raise AttributeError(
+            '{} is not registered as {}.{}'.format(
+                swagger_object_type, self.__module__, self.__name__,
+            ),
+        )
+
+    def get_schema(cls, swagger_object_type):
+        """
+        Extract the JSON schema definition for the gives Swagger2dot0 Swagger object type
+
+        :param swagger_object_type: Swagger2dot0 type to extract the JSON schema
+
+        :return: JSON schema definition
+        :rtype: dict
+        :raises: AttributeError if swagger_object_type is not known
+        """
+        # getattr is called to ensure exception in case swagger_object_type is not registered
+        return cls.mappings[getattr(cls, swagger_object_type)]
+
+    def is_valid(cls, swagger_object_type, object_dict, object_resolver=None):
+        """
+        Checks if `object_dict` if a valid `swagger_object_type` swagger object.
+
+        :param swagger_object_type: Swagger2dot0 type to validate
+        :type swagger_object_type: str
+        :param object_dict: swagger spec json dict to determine the type
+        :type object_dict: dict
+        :param object_resolver: swagger spec reference resolver (ie. return value of ``swagger_spec_validator.validator20.validate_spec``)
+        :type object_resolver: jsonschema.RefResolver
+
+        :return: True if `object_dict` is conform to `swagger_object_type` Schema, False otherwise
+        :rtype: bool
+        :raises: AttributeError if swagger_object_type is not known
+        """
+        try:
+            validator_cls = ref_validators.create_dereffing_validator(object_resolver)
+            validator = validator_cls(
+                cls.get_schema(swagger_object_type),
+                resolver=cls.resolver,
+            )
+            validator.validate(object_dict)
+            return True
+        except ValidationError:
+            return False
+
+
+class Swagger2dot0ObjectType(with_metaclass(_Swagger2dot0ObjectType)):
+    pass
```

Comment:
I remember now why I defined the metaclass :) 
Basically this is needed in order to use ``Swagger2dot0ObjectType`` as an ``Enum`` class without importing ``enum`` (since in python 2.7 is not present by default).

I can:
1. better document this
2. use https://docs.python.org/3/library/enum.html#functional-api and add ``enum34`` dependency

@sjaensch do you have preferences?